### DISCUSSION TITLE
Snips keyboard shortcut creates conflict - Remapping the Snips shortcut.

### DISCUSSION THREAD

**Discussion by raamdev** [blinksh/blink#1895] [Support] [answered]
### Checklist

- [x] Using latest version on the [App Store](http://itunes.apple.com/app/id1156707581)
- [x] Read the [docs](http://blink.sh/docs/)
- [x] Searched for [existing GitHub issues](https://github.com/blinksh/blink/issues)

### Configuration

Blink v17.1.0.854, iOS 17.1.1

### Describe the bug

The Snips feature introduced in v17.0.0 uses the `Command` + `Shift` + `<` key combination, which prevents using that key combination in, for example, Emacs (I use Emacs via Blink over SSH + Mosh and that key combo became unusable after I updated to v17). There doesn't appear to be any way to override the Snips shortcut, or to disable the Snips feature to free up that key combination.

**Comment by carloscabanero** [comment, marked as answer]
Hi! In Config > Keyboard > Blink Shortcuts there is the option to remap Show Snips to something else. Have you tried that?

![image](https://github.com/blinksh/blink/assets/536729/11f06cce-f7a9-41a6-a0b9-7483d9ad7788)

**Comment by raamdev** [comment]
Hi @carloscabanero! I had not seen that shortcut until you pointed it out (thanks!), however I just tried remapping and also deleting the shortcut altogether, but `Command` + `Shift` + `<` now does nothing at all, as if the key combination is not making it through to Emacs. Here are a few things that I just tried after deleting the Blink 'Show Snippets' shortcut:

- Tried completely restarting my iPad
- Tried rebooting my remote linux machine that I SSH into
- Tried connecting to my remote machine via Blink SSH instead of via Blink Mosh

In all cases, the `Command` + `Shift` + `<` shortcut now does nothing at all.

I also use `Command` + `Shift` + `>` in Emacs and that shortcut works fine, but `Command` + `Shift` + `<` stopped working once the Snippets feature was introduced.

I'll also note that while the screenshot that you shared shows `Command` + `Shift` + `,`, the shortcut is actually using `<` since the Shift key is being pressed.

**Comment by carloscabanero** [comment]
Hmmmm... cannot reproduce from my side. Can you do a `showkey -a` and tell me what Cmd-Shift-< is sending when you have that disabled? I get 2c and 2e which is what the sequence is supposed to be.

**Comment by raamdev** [comment]
I get no output at all when I use `showkey -a` and press `Command` + `Shift` + `<`. 

I should note that I'm using a hardware keyboard (the Apple iPad Pro Keyboard). I tried disconnecting the iPad from the keyboard and reconnecting it, but that made no difference.

**Comment by carloscabanero** [comment]
That is weird. Are you maybe using a different layout? I' using an external keyboard as well, and tried with iPad keyboard and my usual mechanical. 

We can reset the keyboard information. Can you do a `mv .blink/kb.json .blink/kb.json.bk`. Then fully restart Blink. Try to set Show Snippets to something else, then test `showkey -a` again.

And one last thing if that fails. Please go to https://jsfiddle.net/carloscabanero/q8rs1ftg/, do the keys combination there in Safari, press copy markdown button and paste here results that you have.

**Comment by raamdev** [comment]
Resetting the keyboard information seems to have done the trick! Thanks so much for helping me figure this out.

**Comment by carloscabanero** [comment]
No worries, glad I could help. Let me move this as a Discussion so others may benefit from it, as it was more of a support issue.
